modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/dnd_table/Cell.js

```diff
@@ -38,6 +38,10 @@ function Cell({
 		if (columnName && heading && !isFixed) {
 			const boundingClientRect = cellRef.current.getBoundingClientRect();
 
+			if (columnName === 'item-actions') {
+				boundingClientRect.width = 100;
+			}
```

Comment:
As I previously mentioned in https://github.com/liferay-frontend/liferay-portal/pull/4038#pullrequestreview-1938258570, I believe we can be smarter than this, we can reasonably estimate size based on font size and number of characters in a label.

This solution increases width in all cases, including the vast majority of cases where there is no visible label.

Reply:
Hello @markocikos 

How can we go about estimating the size if the column width calculation depends on the header width?
We have no context on which button will be the widest and this can change across different pages.
[This was brought up by Daniel.](https://liferay.slack.com/archives/C3JBR21HA/p1710255819396489?thread_ts=1710231619.681329&cid=C3JBR21HA)

Let me know if you can point me in the right direction on making these changes.

Reply:
We cannot calculate it accurately, but we can _estimate_ it, based on font size and number of characters in a label. It'll not be perfect, it won't take into account bunch of things, customer's style overrides can easily break it, but it should still be a lot better than just a fixed width.

Reply:
@markocikos Can you elaborate more where the label size and font can be calculated and added?
I haven’t checked myself as it’s middle of the night for me and didn't want to have to wait another day for an answer if I was still unsure after I checked in the morning.
Truly appreciate the help!

Reply:
I took a look at this and one way you might be able to achieve this is with canvas, I found the solution [here](https://stackoverflow.com/questions/118241/calculate-text-width-with-javascript)

For example:
```js
const canvas = document.createElement("canvas");

function getTextWidth(text, font) {
    const context = canvas.getContext("2d");
    
    context.font = font;
   
    return context.measureText(text).width;
};

// usage

const myElement = document.getElementById('header');

const styles = getComputedStyle(myElement);

getTextWidth(myElement.innerText, styles.font)
```

[See fiddle](https://jsfiddle.net/2a5fLcr3/3/)

Reply:
Hello @bryceosterhaus, truly appreciate you taking a look at this and finding this out!

So we would have to change how the cells are currently sized among other things , correct?

Referring to this comment and Daniel’s explanation of the cell functionality.
https://github.com/liferay-frontend/liferay-portal/pull/4038#discussion_r1532489400

Reply:
> So we would have to change how the cells are currently sized among other things , correct?

Yeah I think that would be the case... I'm not totally sure on the product requirements of that, but you may have to find the "largest" cell and then set the width based on that